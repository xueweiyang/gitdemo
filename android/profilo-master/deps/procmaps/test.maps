00400000-004ef000 r-xp 00000000 fc:01 12616207                           /bin/bash
006ef000-006f0000 r--p 000ef000 fc:01 12616207                           /bin/bash
006f0000-006f9000 rw-p 000f0000 fc:01 12616207                           /bin/bash
006f9000-006ff000 rw-p 00000000 00:00 0 
017a9000-01825000 rw-p 00000000 00:00 0                                  [heap]
7fbca87c9000-7fbca87d4000 r-xp 00000000 fc:01 6293131                    /lib/x86_64-linux-gnu/libnss_files-2.19.so
7fbca87d4000-7fbca89d3000 ---p 0000b000 fc:01 6293131                    /lib/x86_64-linux-gnu/libnss_files-2.19.so
7fbca89d3000-7fbca89d4000 r--p 0000a000 fc:01 6293131                    /lib/x86_64-linux-gnu/libnss_files-2.19.so
7fbca89d4000-7fbca89d5000 rw-p 0000b000 fc:01 6293131                    /lib/x86_64-linux-gnu/libnss_files-2.19.so
7fbca89d5000-7fbca89e0000 r-xp 00000000 fc:01 6293133                    /lib/x86_64-linux-gnu/libnss_nis-2.19.so
7fbca89e0000-7fbca8bdf000 ---p 0000b000 fc:01 6293133                    /lib/x86_64-linux-gnu/libnss_nis-2.19.so
7fbca8bdf000-7fbca8be0000 r--p 0000a000 fc:01 6293133                    /lib/x86_64-linux-gnu/libnss_nis-2.19.so
7fbca8be0000-7fbca8be1000 rw-p 0000b000 fc:01 6293133                    /lib/x86_64-linux-gnu/libnss_nis-2.19.so
7fbca8be1000-7fbca8bf8000 r-xp 00000000 fc:01 6293114                    /lib/x86_64-linux-gnu/libnsl-2.19.so
7fbca8bf8000-7fbca8df7000 ---p 00017000 fc:01 6293114                    /lib/x86_64-linux-gnu/libnsl-2.19.so
7fbca8df7000-7fbca8df8000 r--p 00016000 fc:01 6293114                    /lib/x86_64-linux-gnu/libnsl-2.19.so
7fbca8df8000-7fbca8df9000 rw-p 00017000 fc:01 6293114                    /lib/x86_64-linux-gnu/libnsl-2.19.so
7fbca8df9000-7fbca8dfb000 rw-p 00000000 00:00 0 
7fbca8dfb000-7fbca8e04000 r-xp 00000000 fc:01 6293104                    /lib/x86_64-linux-gnu/libnss_compat-2.19.so
7fbca8e04000-7fbca9003000 ---p 00009000 fc:01 6293104                    /lib/x86_64-linux-gnu/libnss_compat-2.19.so
7fbca9003000-7fbca9004000 r--p 00008000 fc:01 6293104                    /lib/x86_64-linux-gnu/libnss_compat-2.19.so
7fbca9004000-7fbca9005000 rw-p 00009000 fc:01 6293104                    /lib/x86_64-linux-gnu/libnss_compat-2.19.so
7fbca9005000-7fbca92ce000 r--p 00000000 fc:01 6044294                    /usr/lib/locale/locale-archive
7fbca92ce000-7fbca9489000 r-xp 00000000 fc:01 6293122                    /lib/x86_64-linux-gnu/libc-2.19.so
7fbca9489000-7fbca9689000 ---p 001bb000 fc:01 6293122                    /lib/x86_64-linux-gnu/libc-2.19.so
7fbca9689000-7fbca968d000 r--p 001bb000 fc:01 6293122                    /lib/x86_64-linux-gnu/libc-2.19.so
7fbca968d000-7fbca968f000 rw-p 001bf000 fc:01 6293122                    /lib/x86_64-linux-gnu/libc-2.19.so
7fbca968f000-7fbca9694000 rw-p 00000000 00:00 0 
7fbca9694000-7fbca9697000 r-xp 00000000 fc:01 6293141                    /lib/x86_64-linux-gnu/libdl-2.19.so
7fbca9697000-7fbca9896000 ---p 00003000 fc:01 6293141                    /lib/x86_64-linux-gnu/libdl-2.19.so
7fbca9896000-7fbca9897000 r--p 00002000 fc:01 6293141                    /lib/x86_64-linux-gnu/libdl-2.19.so
7fbca9897000-7fbca9898000 rw-p 00003000 fc:01 6293141                    /lib/x86_64-linux-gnu/libdl-2.19.so
7fbca9898000-7fbca98bd000 r-xp 00000000 fc:01 6294567                    /lib/x86_64-linux-gnu/libtinfo.so.5.9
7fbca98bd000-7fbca9abc000 ---p 00025000 fc:01 6294567                    /lib/x86_64-linux-gnu/libtinfo.so.5.9
7fbca9abc000-7fbca9ac0000 r--p 00024000 fc:01 6294567                    /lib/x86_64-linux-gnu/libtinfo.so.5.9
7fbca9ac0000-7fbca9ac1000 rw-p 00028000 fc:01 6294567                    /lib/x86_64-linux-gnu/libtinfo.so.5.9
7fbca9ac1000-7fbca9ae4000 r-xp 00000000 fc:01 6293124                    /lib/x86_64-linux-gnu/ld-2.19.so
7fbca9cb3000-7fbca9cb6000 rw-p 00000000 00:00 0 
7fbca9cda000-7fbca9ce1000 r--s 00000000 fc:01 6705529                    /usr/lib/x86_64-linux-gnu/gconv/gconv-modules.cache
7fbca9ce1000-7fbca9ce3000 rw-p 00000000 00:00 0 
7fbca9ce3000-7fbca9ce4000 r--p 00022000 fc:01 6293124                    /lib/x86_64-linux-gnu/ld-2.19.so
7fbca9ce4000-7fbca9ce5000 rw-p 00023000 fc:01 6293124                    /lib/x86_64-linux-gnu/ld-2.19.so
7fbca9ce5000-7fbca9ce6000 rw-p 00000000 00:00 0 
7fff944b0000-7fff944d1000 rw-p 00000000 00:00 0                          [stack]
7fff945fe000-7fff94600000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
