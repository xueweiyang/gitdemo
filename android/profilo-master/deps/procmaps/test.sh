#!/bin/sh
rt() {
    ./procmaps-test 1 test.maps "$@"
}

echo @@ Maps
rt

echo @@ Expected success
rt 0000000000400000
rt 00000000006ef000
rt 00000000006f0000
rt 00000000006f9000
rt 00000000017a9000
rt 00007fbca87c9000
rt 00007fbca87d4000
rt 00007fbca89d3000
rt 00007fbca89d4000
rt 00007fbca89d5000
rt 00007fbca89e0000
rt 00007fbca8bdf000
rt 00007fbca8be0000
rt 00007fbca8be1000
rt 00007fbca8bf8000
rt 00007fbca8df7000
rt 00007fbca8df8000
rt 00007fbca8df9000
rt 00007fbca8dfb000
rt 00007fbca8e04000
rt 00007fbca9003000
rt 00007fbca9004000
rt 00007fbca9005000
rt 00007fbca92ce000
rt 00007fbca9489000
rt 00007fbca9689000
rt 00007fbca968d000
rt 00007fbca968f000
rt 00007fbca9694000
rt 00007fbca9697000
rt 00007fbca9896000
rt 00007fbca9897000
rt 00007fbca9898000
rt 00007fbca98bd000
rt 00007fbca9abc000
rt 00007fbca9ac0000
rt 00007fbca9ac1000
rt 00007fbca9cb3000
rt 00007fbca9cda000
rt 00007fbca9ce1000
rt 00007fbca9ce3000
rt 00007fbca9ce4000
rt 00007fbca9ce5000
rt 00007fff944b0000
rt 00007fff945fe000
rt ffffffffff600000
rt 00007fbca968d005

echo @@ Expected failure
rt 0
rt 1
rt ffffffffffffffff
rt 00007fbca9cb6000
rt deadf00ddeadf00d
