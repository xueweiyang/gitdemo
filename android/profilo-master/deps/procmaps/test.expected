@@ Maps
00: 0000000000400000 - 00000000004ef000 [r-xp] [/bin/bash]
01: 00000000006ef000 - 00000000006f0000 [r--p] [/bin/bash]
02: 00000000006f0000 - 00000000006f9000 [rw-p] [/bin/bash]
03: 00000000006f9000 - 00000000006ff000 [rw-p] [(null)]
04: 00000000017a9000 - 0000000001825000 [rw-p] [[heap]]
05: 00007fbca87c9000 - 00007fbca87d4000 [r-xp] [/lib/x86_64-linux-gnu/libnss_files-2.19.so]
06: 00007fbca87d4000 - 00007fbca89d3000 [---p] [/lib/x86_64-linux-gnu/libnss_files-2.19.so]
07: 00007fbca89d3000 - 00007fbca89d4000 [r--p] [/lib/x86_64-linux-gnu/libnss_files-2.19.so]
08: 00007fbca89d4000 - 00007fbca89d5000 [rw-p] [/lib/x86_64-linux-gnu/libnss_files-2.19.so]
09: 00007fbca89d5000 - 00007fbca89e0000 [r-xp] [/lib/x86_64-linux-gnu/libnss_nis-2.19.so]
10: 00007fbca89e0000 - 00007fbca8bdf000 [---p] [/lib/x86_64-linux-gnu/libnss_nis-2.19.so]
11: 00007fbca8bdf000 - 00007fbca8be0000 [r--p] [/lib/x86_64-linux-gnu/libnss_nis-2.19.so]
12: 00007fbca8be0000 - 00007fbca8be1000 [rw-p] [/lib/x86_64-linux-gnu/libnss_nis-2.19.so]
13: 00007fbca8be1000 - 00007fbca8bf8000 [r-xp] [/lib/x86_64-linux-gnu/libnsl-2.19.so]
14: 00007fbca8bf8000 - 00007fbca8df7000 [---p] [/lib/x86_64-linux-gnu/libnsl-2.19.so]
15: 00007fbca8df7000 - 00007fbca8df8000 [r--p] [/lib/x86_64-linux-gnu/libnsl-2.19.so]
16: 00007fbca8df8000 - 00007fbca8df9000 [rw-p] [/lib/x86_64-linux-gnu/libnsl-2.19.so]
17: 00007fbca8df9000 - 00007fbca8dfb000 [rw-p] [(null)]
18: 00007fbca8dfb000 - 00007fbca8e04000 [r-xp] [/lib/x86_64-linux-gnu/libnss_compat-2.19.so]
19: 00007fbca8e04000 - 00007fbca9003000 [---p] [/lib/x86_64-linux-gnu/libnss_compat-2.19.so]
20: 00007fbca9003000 - 00007fbca9004000 [r--p] [/lib/x86_64-linux-gnu/libnss_compat-2.19.so]
21: 00007fbca9004000 - 00007fbca9005000 [rw-p] [/lib/x86_64-linux-gnu/libnss_compat-2.19.so]
22: 00007fbca9005000 - 00007fbca92ce000 [r--p] [/usr/lib/locale/locale-archive]
23: 00007fbca92ce000 - 00007fbca9489000 [r-xp] [/lib/x86_64-linux-gnu/libc-2.19.so]
24: 00007fbca9489000 - 00007fbca9689000 [---p] [/lib/x86_64-linux-gnu/libc-2.19.so]
25: 00007fbca9689000 - 00007fbca968d000 [r--p] [/lib/x86_64-linux-gnu/libc-2.19.so]
26: 00007fbca968d000 - 00007fbca968f000 [rw-p] [/lib/x86_64-linux-gnu/libc-2.19.so]
27: 00007fbca968f000 - 00007fbca9694000 [rw-p] [(null)]
28: 00007fbca9694000 - 00007fbca9697000 [r-xp] [/lib/x86_64-linux-gnu/libdl-2.19.so]
29: 00007fbca9697000 - 00007fbca9896000 [---p] [/lib/x86_64-linux-gnu/libdl-2.19.so]
30: 00007fbca9896000 - 00007fbca9897000 [r--p] [/lib/x86_64-linux-gnu/libdl-2.19.so]
31: 00007fbca9897000 - 00007fbca9898000 [rw-p] [/lib/x86_64-linux-gnu/libdl-2.19.so]
32: 00007fbca9898000 - 00007fbca98bd000 [r-xp] [/lib/x86_64-linux-gnu/libtinfo.so.5.9]
33: 00007fbca98bd000 - 00007fbca9abc000 [---p] [/lib/x86_64-linux-gnu/libtinfo.so.5.9]
34: 00007fbca9abc000 - 00007fbca9ac0000 [r--p] [/lib/x86_64-linux-gnu/libtinfo.so.5.9]
35: 00007fbca9ac0000 - 00007fbca9ac1000 [rw-p] [/lib/x86_64-linux-gnu/libtinfo.so.5.9]
36: 00007fbca9ac1000 - 00007fbca9ae4000 [r-xp] [/lib/x86_64-linux-gnu/ld-2.19.so]
37: 00007fbca9cb3000 - 00007fbca9cb6000 [rw-p] [(null)]
38: 00007fbca9cda000 - 00007fbca9ce1000 [r--s] [/usr/lib/x86_64-linux-gnu/gconv/gconv-modules.cache]
39: 00007fbca9ce1000 - 00007fbca9ce3000 [rw-p] [(null)]
40: 00007fbca9ce3000 - 00007fbca9ce4000 [r--p] [/lib/x86_64-linux-gnu/ld-2.19.so]
41: 00007fbca9ce4000 - 00007fbca9ce5000 [rw-p] [/lib/x86_64-linux-gnu/ld-2.19.so]
42: 00007fbca9ce5000 - 00007fbca9ce6000 [rw-p] [(null)]
43: 00007fff944b0000 - 00007fff944d1000 [rw-p] [[stack]]
44: 00007fff945fe000 - 00007fff94600000 [r-xp] [[vdso]]
45: ffffffffff600000 - ffffffffff601000 [r-xp] [[vsyscall]]
@@ Expected success
0000000000400000: 0000000000400000 - 00000000004ef000 [r-xp] [/bin/bash]
00000000006ef000: 00000000006ef000 - 00000000006f0000 [r--p] [/bin/bash]
00000000006f0000: 00000000006f0000 - 00000000006f9000 [rw-p] [/bin/bash]
00000000006f9000: 00000000006f9000 - 00000000006ff000 [rw-p] [(null)]
00000000017a9000: 00000000017a9000 - 0000000001825000 [rw-p] [[heap]]
00007fbca87c9000: 00007fbca87c9000 - 00007fbca87d4000 [r-xp] [/lib/x86_64-linux-gnu/libnss_files-2.19.so]
00007fbca87d4000: 00007fbca87d4000 - 00007fbca89d3000 [---p] [/lib/x86_64-linux-gnu/libnss_files-2.19.so]
00007fbca89d3000: 00007fbca89d3000 - 00007fbca89d4000 [r--p] [/lib/x86_64-linux-gnu/libnss_files-2.19.so]
00007fbca89d4000: 00007fbca89d4000 - 00007fbca89d5000 [rw-p] [/lib/x86_64-linux-gnu/libnss_files-2.19.so]
00007fbca89d5000: 00007fbca89d5000 - 00007fbca89e0000 [r-xp] [/lib/x86_64-linux-gnu/libnss_nis-2.19.so]
00007fbca89e0000: 00007fbca89e0000 - 00007fbca8bdf000 [---p] [/lib/x86_64-linux-gnu/libnss_nis-2.19.so]
00007fbca8bdf000: 00007fbca8bdf000 - 00007fbca8be0000 [r--p] [/lib/x86_64-linux-gnu/libnss_nis-2.19.so]
00007fbca8be0000: 00007fbca8be0000 - 00007fbca8be1000 [rw-p] [/lib/x86_64-linux-gnu/libnss_nis-2.19.so]
00007fbca8be1000: 00007fbca8be1000 - 00007fbca8bf8000 [r-xp] [/lib/x86_64-linux-gnu/libnsl-2.19.so]
00007fbca8bf8000: 00007fbca8bf8000 - 00007fbca8df7000 [---p] [/lib/x86_64-linux-gnu/libnsl-2.19.so]
00007fbca8df7000: 00007fbca8df7000 - 00007fbca8df8000 [r--p] [/lib/x86_64-linux-gnu/libnsl-2.19.so]
00007fbca8df8000: 00007fbca8df8000 - 00007fbca8df9000 [rw-p] [/lib/x86_64-linux-gnu/libnsl-2.19.so]
00007fbca8df9000: 00007fbca8df9000 - 00007fbca8dfb000 [rw-p] [(null)]
00007fbca8dfb000: 00007fbca8dfb000 - 00007fbca8e04000 [r-xp] [/lib/x86_64-linux-gnu/libnss_compat-2.19.so]
00007fbca8e04000: 00007fbca8e04000 - 00007fbca9003000 [---p] [/lib/x86_64-linux-gnu/libnss_compat-2.19.so]
00007fbca9003000: 00007fbca9003000 - 00007fbca9004000 [r--p] [/lib/x86_64-linux-gnu/libnss_compat-2.19.so]
00007fbca9004000: 00007fbca9004000 - 00007fbca9005000 [rw-p] [/lib/x86_64-linux-gnu/libnss_compat-2.19.so]
00007fbca9005000: 00007fbca9005000 - 00007fbca92ce000 [r--p] [/usr/lib/locale/locale-archive]
00007fbca92ce000: 00007fbca92ce000 - 00007fbca9489000 [r-xp] [/lib/x86_64-linux-gnu/libc-2.19.so]
00007fbca9489000: 00007fbca9489000 - 00007fbca9689000 [---p] [/lib/x86_64-linux-gnu/libc-2.19.so]
00007fbca9689000: 00007fbca9689000 - 00007fbca968d000 [r--p] [/lib/x86_64-linux-gnu/libc-2.19.so]
00007fbca968d000: 00007fbca968d000 - 00007fbca968f000 [rw-p] [/lib/x86_64-linux-gnu/libc-2.19.so]
00007fbca968f000: 00007fbca968f000 - 00007fbca9694000 [rw-p] [(null)]
00007fbca9694000: 00007fbca9694000 - 00007fbca9697000 [r-xp] [/lib/x86_64-linux-gnu/libdl-2.19.so]
00007fbca9697000: 00007fbca9697000 - 00007fbca9896000 [---p] [/lib/x86_64-linux-gnu/libdl-2.19.so]
00007fbca9896000: 00007fbca9896000 - 00007fbca9897000 [r--p] [/lib/x86_64-linux-gnu/libdl-2.19.so]
00007fbca9897000: 00007fbca9897000 - 00007fbca9898000 [rw-p] [/lib/x86_64-linux-gnu/libdl-2.19.so]
00007fbca9898000: 00007fbca9898000 - 00007fbca98bd000 [r-xp] [/lib/x86_64-linux-gnu/libtinfo.so.5.9]
00007fbca98bd000: 00007fbca98bd000 - 00007fbca9abc000 [---p] [/lib/x86_64-linux-gnu/libtinfo.so.5.9]
00007fbca9abc000: 00007fbca9abc000 - 00007fbca9ac0000 [r--p] [/lib/x86_64-linux-gnu/libtinfo.so.5.9]
00007fbca9ac0000: 00007fbca9ac0000 - 00007fbca9ac1000 [rw-p] [/lib/x86_64-linux-gnu/libtinfo.so.5.9]
00007fbca9ac1000: 00007fbca9ac1000 - 00007fbca9ae4000 [r-xp] [/lib/x86_64-linux-gnu/ld-2.19.so]
00007fbca9cb3000: 00007fbca9cb3000 - 00007fbca9cb6000 [rw-p] [(null)]
00007fbca9cda000: 00007fbca9cda000 - 00007fbca9ce1000 [r--s] [/usr/lib/x86_64-linux-gnu/gconv/gconv-modules.cache]
00007fbca9ce1000: 00007fbca9ce1000 - 00007fbca9ce3000 [rw-p] [(null)]
00007fbca9ce3000: 00007fbca9ce3000 - 00007fbca9ce4000 [r--p] [/lib/x86_64-linux-gnu/ld-2.19.so]
00007fbca9ce4000: 00007fbca9ce4000 - 00007fbca9ce5000 [rw-p] [/lib/x86_64-linux-gnu/ld-2.19.so]
00007fbca9ce5000: 00007fbca9ce5000 - 00007fbca9ce6000 [rw-p] [(null)]
00007fff944b0000: 00007fff944b0000 - 00007fff944d1000 [rw-p] [[stack]]
00007fff945fe000: 00007fff945fe000 - 00007fff94600000 [r-xp] [[vdso]]
ffffffffff600000: ffffffffff600000 - ffffffffff601000 [r-xp] [[vsyscall]]
00007fbca968d005: 00007fbca968d000 - 00007fbca968f000 [rw-p] [/lib/x86_64-linux-gnu/libc-2.19.so]
@@ Expected failure
0000000000000000: [NULL vma]
0000000000000001: [NULL vma]
ffffffffffffffff: [NULL vma]
00007fbca9cb6000: [NULL vma]
deadf00ddeadf00d: [NULL vma]
